interpreter/value.go

```diff
@@ -159,30 +159,37 @@ type MemberAccessibleValue interface {
 	SetMember(interpreter *Interpreter, locationRange LocationRange, name string, value Value) bool
 }
 
+type ValueComparisonContext interface {
+	common.MemoryGauge
+	ValueStaticTypeContext
```

Comment:
I guess `ValueStaticTypeContext` is included because they are needed for container equality (getting children from containers)?

Reply:
No, this is actually needed for some implementations of the comparison functions, like `Less`, especially in implementations of number value types, like `IntValue`, mainly to get the static type information for producing useful error messages, e.g.:

```go
func (v IntValue) Less(context ValueComparisonContext, other ComparableValue, locationRange LocationRange) BoolValue {
	o, ok := other.(IntValue)
	if !ok {
		panic(InvalidOperandsError{
			Operation:     ast.OperationLess,
			LeftType:      v.StaticType(context),
			RightType:     other.StaticType(context),
			LocationRange: locationRange,
		})
	}
    // ...
```